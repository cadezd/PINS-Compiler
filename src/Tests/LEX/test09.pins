# preprost program
num:integer = 2;
zmnozek:integer = num*num;
sum:integer = 0;

# je to zanka?
for (i:integer = 0; (i > zmnozek) == false; i = i + 1) {
    sum = sum + i;
}