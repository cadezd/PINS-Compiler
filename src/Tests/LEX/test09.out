[2:1-2:3] IDENTIFIER:num
[2:4] OP_COLON::
[2:5-2:11] AT_INTEGER:integer
[2:13] OP_ASSIGN:=
[2:15] C_INTEGER:2
[2:16] OP_SEMICOLON:;
[3:1-3:7] IDENTIFIER:zmnozek
[3:8] OP_COLON::
[3:9-3:15] AT_INTEGER:integer
[3:17] OP_ASSIGN:=
[3:19-3:21] IDENTIFIER:num
[3:22] OP_MUL:*
[3:23-3:25] IDENTIFIER:num
[3:26] OP_SEMICOLON:;
[4:1-4:3] IDENTIFIER:sum
[4:4] OP_COLON::
[4:5-4:11] AT_INTEGER:integer
[4:13] OP_ASSIGN:=
[4:15] C_INTEGER:0
[4:16] OP_SEMICOLON:;
[7:1-7:3] KW_FOR:for
[7:5] OP_LPARENT:(
[7:6] IDENTIFIER:i
[7:7] OP_COLON::
[7:8-7:14] AT_INTEGER:integer
[7:16] OP_ASSIGN:=
[7:18] C_INTEGER:0
[7:19] OP_SEMICOLON:;
[7:21] OP_LPARENT:(
[7:22] IDENTIFIER:i
[7:24] OP_GT:>
[7:26-7:32] IDENTIFIER:zmnozek
[7:33] OP_RPARENT:)
[7:35-7:36] OP_EQ:==
[7:38-7:42] C_LOGICAL:false
[7:43] OP_SEMICOLON:;
[7:45] IDENTIFIER:i
[7:47] OP_ASSIGN:=
[7:49] IDENTIFIER:i
[7:51] OP_ADD:+
[7:53] C_INTEGER:1
[7:54] OP_RPARENT:)
[7:56] OP_LBRACE:{
[8:5-8:7] IDENTIFIER:sum
[8:9] OP_ASSIGN:=
[8:11-8:13] IDENTIFIER:sum
[8:15] OP_ADD:+
[8:17] IDENTIFIER:i
[8:18] OP_SEMICOLON:;
[9:1] OP_RBRACE:}
EOF:EOF